src/cli/tools/wally.ts

```diff
@@ -0,0 +1,72 @@
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { execa } from 'execa';
+import { inject, injectable } from 'inversify';
+import { BaseInstallService } from '../install-tool/base-install.service';
+import {
+  CompressionService,
+  EnvService,
+  HttpService,
+  PathService,
+} from '../services';
+import { getDistro, parse } from '../utils';
+
+@injectable()
+export class WallyInstallService extends BaseInstallService {
+  readonly name = 'wally';
+
+  constructor(
+    @inject(EnvService) envSvc: EnvService,
+    @inject(PathService) pathSvc: PathService,
+    @inject(HttpService) private http: HttpService,
+    @inject(CompressionService) private compress: CompressionService,
+  ) {
+    super(pathSvc, envSvc);
+  }
+
+  override async install(version: string): Promise<void> {
+    const distro = await getDistro();
+    // wally requires libssl3 which is not easily installable on focal
+    if (distro.versionCode === 'focal') {
+      throw new Error(`Unsupported distro: ${distro.versionCode}`);
+    }
+
+    const baseUrl = `https://github.com/UpliftGames/wally/releases/download/v${version}/`;
```

Comment:
use our own pre built binaries, they work on focal

Reply:
Can I remove the distro check in the installer then?

Reply:
yes

Reply:
Done